Move has_vix_disk_lib into vmware plugin as a seed

Fryguy opened the pull request:
This PR also adds the ability for plugins to provide a seed method for their Engine.  By extending the existing seed mechanism to plugins, we can move the has_vix_disk_lib call to the vmware plugin.

@agrare Please review.  Thoughts on this approach?
cc @carbonin 

Merge with https://github.com/ManageIQ/manageiq-providers-vmware/pull/501

Cross repo tests: https://github.com/ManageIQ/manageiq-cross_repo-tests/pull/42

agrare commented:
@carbonin if `seed_rest` did `seed(OTHER_SEEDABLE_CLASSES + seedable_plugin_classes)` or we included the plugin_classes in `OTHER_SEEDABLE_CLASSES` that would get it done right?

carbonin commented:
Yeah, I think that would do it, but I also think @Fryguy wanted to do this differently after talking about it a bit more.

Should have put an update here for that.

carbonin commented:
The issue was that we would be encouraging a monster method in the plugin which would be a catchall for a ton of things that the plugin might want to do to the database.

I think he wanted to make the plugin define a `.seedable_classes` method which we would call here then seed the classes individually like we do for everything in core.

Fryguy commented:
Thanks for reminding me @carbonin cause I actually completely forgot what we discussed 🤣 !

Fryguy commented:
Ok, this should be ready to go now.  I changed it such that plugins provider a seedable_classes method on the Engine, which in turn allows them to enumerate multiple things that can be seeded.  See ManageIQ/manageiq-providers-vmware#501 for an example.

And yeah, I fixed the bug where I actually didn't call the new thing 😁

carbonin commented:
Looks good to me once the plugin change is merged.

agrare commented:
@carbonin Merged the VMware PR
